crates/rpc/rpc/src/eth/cache/mod.rs

```diff
@@ -29,24 +31,35 @@ mod metrics;
 mod multi_consumer;
 pub use multi_consumer::MultiConsumerLruCache;
 
+#[derive(Debug)]
+pub enum ResponseSender {
+    Block(BlockResponseSender),
+    BlockTransactions(BlockTransactionsResponseSender),
+    BlockWithSenders(BlockWithSendersResponseSender),
+}
+
+#[derive(Debug)]
+enum BlockData {
+    Block(Block),
+    BlockWithSenders(BlockWithSenders),
+}
```

Comment:
I think we can simplify this by a lot if we always store `BlockWithSenders` instead, so every missing block should be fetched with senders